apps/scan/frontend/src/components/cast_vote_record_sync_modal.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+import { throwIllegalValue } from '@votingworks/basics';
+import {
+  Button,
+  Loading,
+  Modal,
+  P,
+  useQueryChangeListener,
+} from '@votingworks/ui';
+
+import { exportCastVoteRecordsToUsbDrive, getUsbDriveStatus } from '../api';
+
+type ModalState = 'closed' | 'prompt' | 'syncing' | 'success' | 'error';
+
+export function CastVoteRecordSyncModal(): JSX.Element | null {
```

Comment:
This same code path and UX could be used for recovery after unexpected failures partway through an export operation, which is cool! Planning to add in a follow-up PR